programs/vote/src/vote_state/mod.rs

```diff
@@ -141,6 +141,37 @@ pub fn to<T: WritableAccount>(versioned: &VoteStateVersions, account: &mut T) ->
     VoteState::serialize(versioned, account.data_as_mut_slice()).ok()
 }
 
+// Updates the vote account state with a new VoteState instance.  This is required temporarily during the
+// upgrade of vote account state from V1_14_11 to Current.
+fn set_vote_account_state(
+    vote_account: &mut BorrowedAccount,
+    vote_state: VoteState,
+    feature_set: &FeatureSet,
+) -> Result<(), InstructionError> {
+    // Only if vote_state_add_vote_latency feature is enabled should the new version of vote state be stored
+    if feature_set.is_active(&feature_set::vote_state_add_vote_latency::id()) {
+        // If the account is not large enough to store the vote state, then attempt a realloc to make it large enough.
+        // The realloc can only proceed if the vote account has balance sufficient for rent exemption at the new size.
+        if (vote_account.get_data().len() < VoteState::size_of())
+            && (!vote_account.is_rent_exempt_at_data_length(VoteState::size_of())
```

Comment:
In these cases where these accounts have insufficient lamports, they will receive fewer rewards?

So it will be up to the validator operator to check their vote accounts have sufficient balance after this change is released. Probably need to make some comms around this to allow validators to easily check if their accounts have been upgraded to the new version, and catch cases where they could fail:

1. they have sufficient balance
2. The `!vote_account.set_data_length(VoteState::size_of()).is_ok())` check below didn't silently fail (not sure of all the possibilities here)

Reply:
What will happen is that if the accounts still have insufficient lamports by the time the *second* change is released and its feature enabled, which would actually implement timely vote credits, then yes, those validators will receive a single vote credit per vote, which is the minimum possible.

I believe that this should be easily avoided though because the enablement of the second feature would come a significant time after the enablement of the feature implemented in this current change, giving validators plenty of time to provide sufficient lamports into their vote account to allow the state upgrade.

Reply:
I could log if the `vote_account.set_data_length(VoteState::size_of()` call fails.  That would at least leave some record of the failure easily visible in a place like the solana explorer for validators who want to figure out why they might not be getting updated state.  I didn't do it because I wasn't sure if logging in the vote program would be an issue since its a very frequently run program and I didn't want to increase the storage cost on the explorer for retaining details about the transaction (including the log messages).

Reply:
In addition, once this current change's feature is switched, we can survey vote accounts to see how many haven't upgraded, and only proceed with the timely vote credits feature once all of the (or the vast majority of the) active vote accounts have successfully upgraded to the new state.